sklearn/base.py

```diff
@@ -290,10 +290,11 @@ def __repr__(self):
                                                offset=len(class_name),),)
 
     def __getstate__(self):
```

Comment:
Just wondering whether we should adopt a similar strategy for `__setstate__`? Quickly looking at it, it seems like we are doing `startswith('sklearn.')` there as well to avoid messing with classes outside scikit-learn deriving from `sklearn.base.BaseEstimator`.